src/commands/git-metadata/library.ts

```diff
@@ -0,0 +1,70 @@
+import {newApiKeyValidator} from '../../helpers/apikey'
+import {RequestBuilder} from '../../helpers/interfaces'
+import {upload, UploadOptions, UploadStatus} from '../../helpers/upload'
+import {getRequestBuilder} from '../../helpers/utils'
+import {getCommitInfo, newSimpleGit} from './git'
+import {CommitInfo} from './interfaces'
+
+export const shouldAddSourceCodeIntegration = async (apiKey: string | undefined): Promise<boolean> => {
+  try {
+    const simpleGit = await newSimpleGit()
+    const isRepo = simpleGit.checkIsRepo()
+
+    // Only enable if the system has `git` installed and we're in a git repo
+    return apiKey !== undefined && isRepo
+  } catch {
+    return false
+  }
+}
```

Comment:
nit: Just wondering if having `apiKey !== undefined` is really necessary in this exported func since it's passed as arg. Shouldn't it be checked by whoever call this function?  
Without it the function could be renamed to something like `isCwdInGitRepository` making what is being done explicit.